File: src/common/enums/sse.ts
/**
 * SSE event type discriminator (Vercel AI SDK–aligned, Cursor-style).
 */
export enum SSEEventType {
  Session = "session",
  Status = "status",
  TextDelta = "text-delta",
  TextEnd = "text-end",
  ToolCall = "tool-call",
  ToolResult = "tool-result",
  ApprovalRequested = "approval-requested",
  Error = "error",
  Finish = "finish",
}

/**
 * Status code for "status" events (planning | thinking | executing | timeout).
 */
export enum StatusCode {
  Planning = "planning",
  Thinking = "thinking",
  Executing = "executing",
  Timeout = "timeout",
}

/**
 * Finish reason for "finish" events.
 */
export enum FinishReason {
  Stop = "stop",
  Length = "length",
  ToolCall = "tool-call",
  ContentFilter = "content-filter",
  Error = "error",
  Abort = "abort",
}

/**
 * Stream trigger type: what initiated the stream (message, approval, reject, etc.).
 */
export enum StreamTriggerType {
  Message = "message",
  System = "system",
  Tool = "tool",
  Approve = "approve",
  Reject = "reject",
  Skip = "skip",
}

File: app/common/enums/sse.py
"""SSE event enums (Vercel AI SDK–aligned, Cursor-style)."""

from enum import Enum


class SSEEventType(str, Enum):
    """SSE event type discriminator."""

    SESSION = "session"
    STATUS = "status"
    TEXT_DELTA = "text-delta"
    TEXT_END = "text-end"
    TOOL_CALL = "tool-call"
    TOOL_RESULT = "tool-result"
    APPROVAL_REQUESTED = "approval-requested"
    ERROR = "error"
    FINISH = "finish"


class StatusCode(str, Enum):
    """Status code for 'status' events."""

    THINKING = "thinking"
    EXECUTING = "executing"
    TIMEOUT = "timeout"


class StatusMessage(str, Enum):
    """Human-readable status messages."""

    PLANNING = "Planning"
    APPLYING = "Applying"
    CANCELLING = "Cancelling"


class FinishReason(str, Enum):
    """Finish reason for 'finish' events."""

    STOP = "stop"
    LENGTH = "length"
    TOOL_CALL = "tool-call"
    CONTENT_FILTER = "content-filter"
    ERROR = "error"
    ABORT = "abort"
